frontend/javascripts/oxalis/model/sagas/settings_saga.ts

```diff
@@ -36,6 +36,7 @@ function* pushDatasetSettingsAsync(): Saga<void> {
       SETTINGS_RETRY_DELAY,
       updateDatasetConfiguration,
       dataset,
+      // mask updates to layer visibility here if this is an annotation?
```

Comment:
I agree that both options are less than beautiful 🤔 I’d say front-end masking/merging that into the request makes sense. The back-end api request is about the dataset configuration, which should stay constant. 

We could also implement the logic in the back-end that fields missing from the request are just not overwritten in the back-end, but this does not work if there are any fields which are actually supposed to be settable to null/none. Also it would require a bit of work in the back-end since it currently just stores the json in the db, just as it is sent by the frontend.

So I’d say if it is feasible to patch this into the request in the front-end, that’s probably the best option.